hyperspy/drawing/mpl_he.py

```diff
@@ -175,6 +181,21 @@ def plot(self, **kwargs):
             self.plot_navigator(**kwargs.pop('navigator_kwds', {}))
         self.plot_signal(**kwargs)
 
+        if "ipympl" in mpl.get_backend():
```

Comment:
This should be used for the notebook backend too. It would need to be optional too.

Reply:
This doesn't work for the notebook backend, afaik (by notebook backend, I mean `%matplotlib notebook`). The reason is that the with the notebook backend, the `figure.canvas` is not a widget. 

```python
%matplotlib notebook
# %matplotlib widget

import matplotlib.pyplot as plt
from ipywidgets.widgets import Output

o = Output()
with o:
    plt.figure()
    plt.plot([1,2,3])
    
display(o)
```

Reply:
The following works fine:
```python
%matplotlib notebook

import matplotlib.pyplot as plt
from ipywidgets.widgets import Output

o = Output()
display(o)
with o:
    plt.figure()
    plt.plot([1,2,3])
```

Can it be adapted to get it to work with the notebook backend?

Reply:
Maybe. Just to note, the following does not work:
```python
%matplotlib notebook

import matplotlib.pyplot as plt
from ipywidgets.widgets import Output, HBox

o = Output()
o2 = Output()
with o:
    plt.figure()
    plt.plot([1,2,3])
with o2:
    plt.figure()
    plt.plot([3,2,1])

display(HBox([o, o2]))
    ```

Reply:
Weird. It sorta does work, but only if I place the display statement before the `with` statements.

Additionally, the layout css does not have any effect:
```python
%matplotlib notebook

import matplotlib.pyplot as plt
from ipywidgets.widgets import Output, HBox

o = Output()
o2 = Output()
margin = "auto 0px auto 0px"
o.layout.margin = margin
o2.layout.margin = margin

display(HBox([o, o2]))
with o:
    plt.figure()
    plt.plot([1,2,3])
with o2:
    plt.figure()
    plt.plot([3,2,1])
```